src_c/display.c

```diff
@@ -1915,29 +1920,29 @@ pg_set_caption(PyObject *self, PyObject *arg)
     _DisplayState *state = DISPLAY_MOD_STATE(self);
     SDL_Window *win = pg_GetDefaultWindow();
     char *title, *icontitle = NULL;
-    if (!PyArg_ParseTuple(arg, "es|es", "UTF-8", &title, "UTF-8", &icontitle))
+
+#ifdef _MSC_VER
+    /* MSVC static analyzer false alarm: assure title is NULL-terminated by
+     * making analyzer assume it was initialised */
+    __analysis_assume(title = "inited");
+#endif
+
+    if (!PyArg_ParseTuple(arg, "s|s", &title, &icontitle))
         return NULL;
 
     if (state->title)
         free(state->title);
+
     state->title = (char *)malloc((strlen(title) + 1) * sizeof(char));
     if (!state->title) {
-        PyErr_NoMemory();
-        goto error;
+        return PyErr_NoMemory();
     }
     strcpy(state->title, title);
     if (win)
         SDL_SetWindowTitle(win, title);
-    /* TODO: icon title? */
 
-    PyMem_Free(title);
-    PyMem_Free(icontitle);
+    /* TODO: icon title? */
     Py_RETURN_NONE;
-
-error:
-    PyMem_Free(title);
```

Comment:
I just had to check this was correct... it is.

https://docs.python.org/3/c-api/arg.html
> In general, when a format sets a pointer to a buffer, the buffer is managed by the corresponding Python object, and the buffer shares the lifetime of this object. You won’t have to release any memory yourself. The only exceptions are es, es#, et and et#.